src/zabapgit_object_samc.prog.abap

```diff
@@ -292,20 +292,24 @@ CLASS lcl_object_samc IMPLEMENTATION.
 
     ro_persistence = mo_persistence.
 
-  ENDMETHOD.
+  ENDMETHOD.                    "get_persistence
 
 
   METHOD lock.
 
-    DATA: objname    TYPE trobj_name,
-          object_key TYPE seu_objkey,
-          objtype    TYPE trobjtype.
+    DATA: objname        TYPE trobj_name,
+          object_key     TYPE seu_objkey,
+          objtype        TYPE trobjtype.
 
     objname    = ms_item-obj_name.
     object_key = ms_item-obj_name.
     objtype    = ms_item-obj_type.
 
-    get_persistence( )->lock(
+    IF mo_persistence IS NOT BOUND.
+      get_persistence( ).
+    ENDIF.
```

Comment:
perhaps call get_persistence( ) in all cases, inside the get_persistence method there is an IF (note this comment applies for multiple places in this PR)